README.md

```diff
@@ -0,0 +1,108 @@
+ - This README file.
+
+### ARM
+
+Each ARM firmware library is organized in the following structure:
+
+```
+.
+└── gd32xxx
+    ├── cmsis
+    │   └── gd
+    │       └── gd32xxx
+    └── standard_peripheral
+        ├── include
+        └── source
+```
+
+### RISC-V
+
+The RISC-V structure is not defined yet. The riscv sub-directory will change
+based on how [NMSIS library](https://github.com/Nuclei-Software/NMSIS/) will
+be implemented in Zephyr. At present date, the structure only reflects current
+manufacturer version.
+
+```
+└── gd32bf103
+    ├── riscv
+    │   ├── drivers
+    │   ├── env_eclipse
+    │   ├── env_iar
+    │   └── stubs
+    └── standard_peripheral
+        ├── include
+        └── source
+```
+
+
+ Any contribution should follow the `How to submit code` using as premisses the
+ [Zephyr Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/index.html).
+
+# How to submit code
+
+ - When submiting an updated firmware library version, it is important make
+   sure that the last firmware version will be merged. The library version can
+   be checked at [GigaDevice MCU Site](http://www.gd32mcu.com/en/download/).
+ - This repository is managed mainly on Linux. This means that changes should
+   be submited using Linux LF format. Usually running a similar command to
+   `find * type f -print0 | xargs -0 dos2unix -k` should be enough.
+ - Make sure directory structure is respected.
+ - Directory names should be converted to lowercase.
+ - Run `gd32_api_check` script to make sure gd32 public API is conformance.
+   Adjusts should be made otherwise code can be reject. Exceptions should be
+   discussed at review phase.
+
+## The gd32 standard peripheral API
+
+The gd32 standard peripheral API was changed to avoid that macros, enum values
+and function names colide with content from both ZephyrRTOS core and libraries.
+
+The are two general rules:
+
+ - Public API funtions must be prefixed with `gd32_`.
+ - Public defines and enum values must be uppercase and prefixed with `GD32_`.
+
+The cmsis and riscv contents are not part of standard peripheral library and
+changes are treated as exceptions. This means that `<firmware_library>.h>` and
+`system_<firmware_library>.h>` files should reside on that places.
+
+### Grace Period
+
+Rework gd32 standard peripheral API is a huge step. To allow developers move
+forward a grace period was created. The grace period is available until
+`gd32_api_convert` script be available. When the script be available any new
+firmware library must be converted.
+
+Besides there is a grace period developers are obligated to solve any existent
+conflict. The conflict should follow gd32 standard peripheral API rules.
+
+The current know conflict are listed below with respective correction:
+
+ - `<firmware_library>_timer.h/c`: `timer_init()` should be converted to
+   `gd32_timer_init()`.
+ - `<firmware_library>_can.h/c`, all `CAN_` defines should be converted to
+   `GD32_CAN_`.
```

Comment:
Since the script is not yet there, I'd suggest to explain:

1. Current API issues (ie no namespacing)
2. Rules to apply when this happens, and list of known cases.
3. Mention that a script will likely be created to automate conversion of all public API.

Reply:
Sure!